core/domain/platform_parameter_domain.py

```diff
@@ -357,9 +390,12 @@ def _match_version_expression(self, op, value, client_version):
             return False
 
         match = APP_VERSION_WITH_HASH_REGEXP.match(client_version)
-        client_version_without_hash = match.group(1)
-
-        is_equal = value == client_version_without_hash
+        client_version_without_hash = match.group(1) #type: ignore[union-attr] 
```

Comment:
We have added the type ignore because item "None" has no attribute group, indeed we have realised the Mypy test the case where client_version_without_hash is none, even if we assure the line before that it is not None.

Reply:
Would `if not isinstance(match, NoneType)` help here?

Reply:
We cannot use your method because , we validate the condition to enter the if (ie match is not none) but still mypy check with the atribute as a none

Reply:
Done